### ISSUE TITLE
BC break: circle layer is drawn in different z-index

### ISSUE THREAD

**Issue by hrach** [mapbox/mapbox-gl-native#14844]
Hi, after upgrading to Android-8.1.0-alpha.1 from 7.4 our simple CircleLayer-s are drawn behind 3D building. SymbolLayer with images and icon stays on top.

### Steps to reproduce

 1.
 ```
mapStyle.addLayer(circleLayer)
mapStyle.addLayer(markerLayer)
```
no other modification of laers, their z-index, etc.
 2. Upgrade to `com.mapbox.mapboxsdk:mapbox-android-sdk:8.1.0-alpha.1`

### Expected behavior

![image](https://user-images.githubusercontent.com/284263/58891841-83e98080-86ed-11e9-83a2-5e8bf9979c4d.png)

### Actual behavior

![image](https://user-images.githubusercontent.com/284263/58891649-366d1380-86ed-11e9-984c-b8b2154fa458.png)

### Configuration

**Android versions**: Android Q
**Device models:** Pixel 2 
**Mapbox SDK versions:** 8.1.0-alpha.1

**Comment by tobrun** [comment]
cc @tmpsantos

**Comment by hrach** [comment]
This issue is already present in 8.0.1, so we are blocked from getting fix for https://github.com/mapbox/mapbox-gl-native/pull/14890

**Comment by camelCaseSucks** [comment]
I came here to report this. Example attached.

![mapbox-7 3 0](https://user-images.githubusercontent.com/47312363/59557000-40abcf00-8f83-11e9-9590-5205717585a1.png)
![mapbox-8 0 1](https://user-images.githubusercontent.com/47312363/59557001-40abcf00-8f83-11e9-8b02-3620a6e01836.png)


[MapboxTest.tar.gz](https://github.com/mapbox/mapbox-gl-native/files/3293612/MapboxTest.tar.gz)

**Comment by kkaefer** [comment]
This was probably broken by https://github.com/mapbox/mapbox-gl-native/pull/14526

**Comment by hrach** [comment]
Hi, I'd like to ask if there is any eta - e.g. days/weeks/months. Because if it takes longer, we will upgrade the app regardless this bug.

**Comment by tmpsantos** [comment]
> Hi, I'd like to ask if there is any eta - e.g. days/weeks/months. Because if it takes longer, we will upgrade the app regardless this bug.

I can't give you a concrete ETA, but I don't see bandwidth available in our team to work on this issue for the next two weeks.

**Comment by chloekraw** [comment]
> Hi, I'd like to ask if there is any eta - e.g. days/weeks/months. Because if it takes longer, we will upgrade the app regardless this bug.

@hrach, we are aiming to get a fix in the next release (5.2 on iOS / 8.2 on Android), so weeks. thanks for your patience.

**Comment by chloekraw** [comment]
@mapbox/maps-ios @mapbox/maps-android, we should also release patches of iOS 5.0, 5.1 and Android 8.0, 8.1 when this is fixed.
